src/Nox.Core/Domain/IRepository.cs

```diff
@@ -1,8 +1,17 @@
 ﻿
+using System.Linq.Expressions;
+
 namespace Nox.Domain;
 
 public interface IRepository
 {
+    // needs to work with IEntity, IOwnedEntity and localized entity
+    IQueryable<T> Query<T>(Expression<Func<T, bool>> predicate) where T : class, IEntity;
+    IQueryable<T> QueryOwned<T>(Expression<Func<T, bool>> predicate) where T : class, IOwnedEntity;
```

Comment:
we  do not need the Owned, I guess it will not work since we do not have any set registered for owned